src/parser/TurtleParser.cpp

```diff
@@ -453,6 +453,81 @@ bool TurtleParser<T>::rdfLiteral() {
   }
 }
 
+// ______________________________________________________________________
+template <class T>
+TripleComponent TurtleParser<T>::literalAndDatatypeToTripleComponent(
+    std::string_view normalizedLiteralContent,
+    const TripleComponent::Iri& typeIri) {
```

Comment:
I think if you move the implementation to an 
`...Impl`function that additionally takes a reference or pointer to a (possibly templated or inherited) parser,
Then you can also use this same function for the `actual` turtle parsing. Currently, as I see, you introduce a lot of code duplication.